model/dce-poll.cc

```diff
@@ -208,7 +208,7 @@ int dce_select (int nfds, fd_set *readfds, fd_set *writefds, fd_set *exceptfds,
 
   if (timeout)
     {
-      pollTo = timeout->tv_sec * 1000 + timeout->tv_usec / 1000;
+      pollTo = timeout->tv_sec * 1000000 + timeout->tv_usec / 1000;
```

Comment:
I believe we should make the minimum resolution in UtilsAdvanceTime a global one
`current->process->manager->Wait (Time (MicroSeconds (1)));` and either round or warn if this value is smaller than the minimum resolution. Would that solve the issue ?

Reply:
I am now just reading the thread carefully for the first time.  I had some off list discussion with Trish and she does not have a small testcase to contribute.

It seems like there are two issues. First, when dce_select() calls dce_poll() with a non-zero timeout but less than 1ms, and second, when some application calls poll() with timeout of zero (and there needs to be a forced timestep to prevent infinite looping).

For the second issue, if I understand correctly, Matt is proposing to pull out this magic value into a global so that it could be tuned by users more easily, and also round up to 1 timestep if the time set in the global is less than the resolution?? (e.g. if simulation timestep is 1ns and user configures the magic value to picoseconds)

For the first issue, it seems like the issue is when dce_select calls dce_poll but wants sub-millisecond resolution.  Here, I suggest possibly to create a dce_poll_internal method that replaces 'int timeout' with 'ns3::Time timeout', and have dce_select call this variant directly, and have the existing dce_poll() forward to this internal dce_poll_internal().